### PR TITLE
Replace rules with '@typescript-eslint' version and disable React prop types rules

### PR DESCRIPTION
**Pull request by shpakkdv**
Replace rules with '@typescript-eslint' version (default-param-last, no-loss-of-precision) and disable React prop types rules (react/prop-types, react/default-props-match-prop-types)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -10,8 +10,14 @@ module.exports = {
...
   rules: {
+    // The rule is enabled in Airbnb config, but is checked by the TypeScript compiler
+    // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/default-props-match-prop-types.md
+    'react/default-props-match-prop-types': 'off',
     // Append 'tsx' to Airbnb 'react/jsx-filename-extension' rule
     // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-filename-extension.md
     'react/jsx-filename-extension': ['error', { extensions: ['.jsx', '.tsx'] }],
+    // The rule is enabled in Airbnb config, but is checked by the TypeScript compiler
+    // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prop-types.md
+    'react/prop-types': 'off',
```
Comment (iamturns) [inline index.js:21]: These rules also apply to JavaScript files, and we'd want to keep the rules enabled for that

Also some people like to use prop types with TypeScript for runtime type checking

It'd be good to revert these changes, I'm not sure they belong within this config

**Review by iamturns — CHANGES_REQUESTED** [review]
Hey, thanks for the PR!

Great catch on these new rules. I checked the changelog from the latest upgrade, but didn't check the actual code diff 😅

Just a small change requested and then it's ready to merge. Thanks again :)

File: index.js (hunk 1 above)
Comment (shpakkdv) [reply index.js]: Got it! Reverted!

**Comment by iamturns** [comment]
Thanks @shpakkdv !
